presto-main/src/main/java/com/facebook/presto/operator/HashBuilderOperator.java

```diff
@@ -629,6 +655,16 @@ public void close()
         lookupSourceSupplier = null;
         state = State.CLOSED;
         finishMemoryRevoke = finishMemoryRevoke.map(ifPresent -> () -> {});
+        if (unspillInProgress.isPresent()) {
```

Comment:
> Could you please elaborate more on why the finishLookupSourceUnspilling reference cannot be set to Optional.empty() right after the Queue<Page> pages = new ArrayDeque<>(unspilledPages);?

in isBlocked(),   state INPUT_UNSPILLING expects this Optional to be exist.
`case INPUT_UNSPILLING:
                return unspillInProgress.orElseThrow(() -> new IllegalStateException("Unspilling in progress, but unspilling future not set"));`

Reply:
> by design cancelling background spills should be the spiller responsibility

I also fixed it in spiller code. In case in operator close() method,  spiller is forgot to be closed,  cancelling spilling task will be useful.

Reply:
From what I understand the `isBlocked` cannot be called when any other method is executed, as the Operator interface is single threaded. Is not not the case?